### PR TITLE
Fix absolute carbon emission calculus and add discharge to total (denominator) values

### LINKED ISSUE #844: Tooltip always compute wrongly: "0% emission in ... comes from ..."
fullscreen_10_11_2017__00_15

### PR DESCRIPTION
**Pull request by brunolajoie**
- added totalDischarge and totalCo2Discharge to the totalPositive value
- fix error of value for absolute emission per fuel
fix #844

### CODE REVIEW

File: web/app/helpers/tooltip.js [hunk 1]
```diff
@@ -45,10 +44,12 @@ module.exports.showProduction = function(tooltipInstance, mode, country, display
 
     if (displayByEmissions) {
         if (!isStorage) {
-            value *= co2intensity * 1000.0;
+            value *= co2intensity * 1000.0; // MW * gCO2/kWh * 1000 --> gCO2/h
         }
     }
 
+    var absValue = Math.abs(value)
```
Comment (brunolajoie) [inline web/app/helpers/tooltip.js]: absValue needed to be defined *after* value has been updated

File: web/app/helpers/tooltip.js [hunk 2]
```diff
@@ -45,10 +44,12 @@ module.exports.showProduction = function(tooltipInstance, mode, country, display
 
     if (displayByEmissions) {
         if (!isStorage) {
-            value *= co2intensity * 1000.0;
+            value *= co2intensity * 1000.0; // MW * gCO2/kWh * 1000 --> gCO2/h
```
Comment (corradio) [inline web/app/helpers/tooltip.js]: 👍

File: web/app/helpers/tooltip.js (hunk 1 above)
Comment (corradio) [reply web/app/helpers/tooltip.js]: nice catch!!

File: web/app/helpers/tooltip.js [hunk 3]
```diff
@@ -61,12 +62,12 @@ module.exports.showProduction = function(tooltipInstance, mode, country, display
...
     tooltip.select('#capacity-factor-detail').html(
         (format(absValue) || '?') + ' ' +
-        ' / ' + 
+        ' / ' +
         (hasCapacity && format(capacity) || '?'));
 
     var totalPositive = displayByEmissions ?
-        (country.totalCo2Production + country.totalCo2Import) :
-        (country.totalProduction + country.totalImport);
+        (country.totalCo2Production + country.totalCo2Discharge +country.totalCo2Import) : // gCO2eq/h
```
Comment (corradio) [inline web/app/helpers/tooltip.js]: Lacking space between `+` and `country`

File: web/app/helpers/tooltip.js [hunk 4]
```diff
@@ -61,12 +62,12 @@ module.exports.showProduction = function(tooltipInstance, mode, country, display
...
+        (country.totalProduction + country.totalDischarge + country.totalImport);
```
Comment (corradio) [inline web/app/helpers/tooltip.js]: 👍

**Comment by corradio** [comment]
Fix #844 (unless I'm wrong, can you confirm @brunolajoie ?)

File: web/app/helpers/tooltip.js (hunk 3 above)
Comment (brunolajoie) [reply web/app/helpers/tooltip.js]: fixed

**Comment by brunolajoie** [comment]
@corradio ready to merge

**Pull request merged**